spec/models/response_spec.rb

```diff
@@ -97,4 +97,22 @@
       expect(r_previous.checked_out_by_id).to be_nil
     end
   end
+
+  context "responses count" do
+    let!(:user) { create(:user) }
+    let!(:form) { create(:form, question_types: %w[integer]) }
+    let!(:response) { create(:response, user: user, form: form, answer_values: %w[1]) }
+
+    describe "updates response count correctly" do
+      it "on response deletion" do
+        response.destroy
+        expect(form.responses_count).to eq(0)
```

Comment:
There is one key issue here.. You should add `reload` after `form`. If you do, I think you'll find that the spec fails. That's because you're not saving the form in the after_destroy callback.

This is one of the most important things to consider when doing ActiveRecord development and when writing specs for it.

Sometimes it is even best to test before AND after a reload. So I would leave this line and add a second one with a test of the reloaded value.

This stuff isn't easy but you'll get it eventually. Just always try to keep in mind what's in memory vs. what's persisted to the DB.